src/Castle.Core/DynamicProxy/PersistentIsolatedTypesModuleScope.cs

```diff
@@ -54,6 +54,9 @@ public PersistentIsolatedTypesModuleScope(bool savePhysicalAssemblies = true,
 			usingStrongNamedModule = false;
 		}
 
+		// for testing purposes
+		internal event Action<string>? AssemblySaved;
```

Comment:
I don't particularly like adding such stuff only for internal testing. Perhaps it would be worth thinking about doing this in a way that would make sense for public use, too?